app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -25,8 +25,18 @@ def complete
     authenticator = self.class.find_authenticator(params[:provider])
     provider = DiscoursePluginRegistry.auth_providers.find { |p| p.name == params[:provider] }
 
-    if authenticator.can_connect_existing_user? && current_user
+    if session.delete(:auth_reconnect) && authenticator.can_connect_existing_user? && current_user
+      # If we're reconnecting, don't actually try and log the user in
       @auth_result = authenticator.after_authenticate(auth, existing_account: current_user)
+      if (provider && provider.full_screen_login) || cookies['fsl']
```

Comment:
```suggestion
      if provider&.full_screen_login || cookies['fsl']
```

Reply:
Done, and changed further down the file as well